staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -151,69 +152,48 @@ func (a *atomicLastError) Load() error {
 }
 
 func newETCD3Check(c storagebackend.Config, timeout time.Duration, stopCh <-chan struct{}) (func() error, error) {
-	// constructing the etcd v3 client blocks and times out if etcd is not available.
-	// retry in a loop in the background until we successfully create the client, storing the client or error encountered
-
-	lock := sync.RWMutex{}
-	var prober *etcd3ProberMonitor
-	clientErr := fmt.Errorf("etcd client connection not yet established")
-
-	go wait.PollImmediateUntil(time.Second, func() (bool, error) {
-		lock.Lock()
-		defer lock.Unlock()
-		newProber, err := newETCD3ProberMonitor(c)
-		// Ensure that server is already not shutting down.
-		select {
-		case <-stopCh:
-			if err == nil {
-				newProber.Close()
-			}
-			return true, nil
-		default:
-		}
-		if err != nil {
-			clientErr = err
-			return false, nil
-		}
-		prober = newProber
-		clientErr = nil
-		return true, nil
-	}, stopCh)
+	// Create a new prober monitor.
+	prober, err := newETCD3ProberMonitor(c)
+	if err != nil {
+		return nil, err
```

Comment:
Now the function might exit early, so I double checked when `newETCD3ProberMonitor` returns error. There are 2 potential sources of errors: egressLookup and client constructor. Both look fine.